From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: ViKing <viking.yfy@qq.com>
Date: Sat, 29 Aug 2026 17:26:00 +0800
Subject: [PATCH] net: ethernet: qualcomm: release egress VP on prelock error

The flow setup path can allocate an egress virtual port before acquiring
the flowtable lock.  Release that reference together with the ingress VP
when setup fails before the locked cleanup path.

Signed-off-by: ViKing <viking.yfy@qq.com>
---
 drivers/net/ethernet/qualcomm/ppe/ppe_flowtable.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/net/ethernet/qualcomm/ppe/ppe_flowtable.c b/drivers/net/ethernet/qualcomm/ppe/ppe_flowtable.c
--- a/drivers/net/ethernet/qualcomm/ppe/ppe_flowtable.c
+++ b/drivers/net/ethernet/qualcomm/ppe/ppe_flowtable.c
@@ -2866,6 +2866,8 @@
 		dev_put(tdev);
 	goto err_free;
 err_prelock_ivp:
+	if (flow->vp >= 0)
+		ppe_vp_put(ft->ppe_dev, flow->vp);
 	if (flow->ivp >= 0)
 		ppe_vp_put(ft->ppe_dev, flow->ivp);
 	if (tdev)
-- 
2.39.5
